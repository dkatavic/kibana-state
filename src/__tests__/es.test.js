const elasticsearch = require('elasticsearch')
const { MOCK_DASHBOARD } = require('../__mocks__/es-mock-state')
const { addItem, getEsState, removeItem, updateItem } = require('../es')

const mockClient = new elasticsearch.Client()

test('getEsState filters search results and excludes config type hits', async () => {
  expect(await getEsState(mockClient, {})).toMatchSnapshot()
  expect(mockClient.search.mock.calls[0][0]).toMatchSnapshot()
})

test('addItem calls esClient with correct params', async () => {
  const createSpy = jest.spyOn(mockClient, 'create')
  await addItem(mockClient, MOCK_DASHBOARD)
  expect(createSpy).toHaveBeenCalled()
  expect(mockClient.create.mock.calls[0][0]).toMatchSnapshot()
})

test('removeItem calls esClient with correct params', async () => {
  const deleteSpy = jest.spyOn(mockClient, 'delete')
  await removeItem(mockClient, MOCK_DASHBOARD)
  expect(deleteSpy).toHaveBeenCalled()
  expect(mockClient.delete.mock.calls[0][0]).toMatchSnapshot()
})

test('updateItem calls esClient with correct params', async () => {
  const updateSpy = jest.spyOn(mockClient, 'index')
  await updateItem(mockClient, MOCK_DASHBOARD)
  expect(updateSpy).toHaveBeenCalled()
  expect(mockClient.index.mock.calls[0][0]).toMatchSnapshot()
})
