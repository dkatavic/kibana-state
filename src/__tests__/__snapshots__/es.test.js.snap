// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addItem calls esClient with correct params 1`] = `
Object {
  "body": Object {
    "config": Object {
      "tr-checksum": "5332d14e81762116caaf18a82ae1b1753ff91781",
    },
    "index-pattern": Object {
      "fields": "[{\\"name\\":\\"AA\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"RA\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"RD\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"TC\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"TTLs\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"_id\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":false},{\\"name\\":\\"_index\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":false},{\\"name\\":\\"_score\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":false,\\"aggregatable\\":false,\\"readFromDocValues\\":false},{\\"name\\":\\"_source\\",\\"type\\":\\"_source\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":false,\\"aggregatable\\":false,\\"readFromDocValues\\":false},{\\"name\\":\\"_type\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":false},{\\"name\\":\\"answers\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"auth\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"id.orig_h\\",\\"type\\":\\"ip\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"id.orig_p\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"id.resp_h\\",\\"type\\":\\"ip\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"id.resp_p\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"proto\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"qclass\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"qclass_name\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"qtype\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"qtype_name\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"query\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"rcode\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"rcode_name\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"rejected\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"saw_query\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"saw_reply\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"total_answers\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"total_replies\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"trans_id\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"ts\\",\\"type\\":\\"date\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"uid\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true}]",
      "timeFieldName": "ts",
      "title": "metadata_dnsrecordinfo*",
    },
    "type": "index-pattern",
    "updated_at": "2018-04-10T17:14:25.905Z",
  },
  "id": "index-pattern:9d8143d0-3ce2-11e8-80cc-950e99edcb04",
  "index": ".kibana",
  "type": "doc",
}
`;

exports[`getEsState filters search results and excludes config type hits 1`] = `
Object {
  "hits": Object {
    "hits": Array [
      Object {
        "_id": "config:6.2.2",
        "_index": ".kibana",
        "_score": 1,
        "_source": Object {
          "config": Object {
            "buildNum": 16588,
            "defaultIndex": "a7480050-3118-11e8-b629-33a5d6f661b4",
            "tr-version": "1.0.1",
          },
          "type": "config",
          "updated_at": "2018-04-25T08:56:45.446Z",
        },
        "_type": "doc",
      },
      Object {
        "_id": "dashboard:host_dashboard",
        "_index": ".kibana",
        "_score": 1,
        "_source": Object {
          "config": Object {
            "tr-checksum": "00d8211f109fd19f9d84a39523b9bdd716b8c3b5",
          },
          "dashboard": Object {
            "description": "",
            "hits": 0,
            "kibanaSavedObjectMeta": Object {
              "searchSourceJSON": "{\\"query\\":{\\"language\\":\\"lucene\\",\\"query\\":\\"\\"},\\"filter\\":[],\\"highlightAll\\":true,\\"version\\":true}",
            },
            "optionsJSON": "{\\"darkTheme\\":false,\\"hidePanelTitles\\":false,\\"useMargins\\":true}",
            "panelsJSON": "[{\\"panelIndex\\":\\"4\\",\\"gridData\\":{\\"x\\":0,\\"y\\":0,\\"w\\":6,\\"h\\":3,\\"i\\":\\"4\\"},\\"id\\":\\"fca79400-44cf-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"5\\",\\"gridData\\":{\\"x\\":6,\\"y\\":0,\\"w\\":6,\\"h\\":3,\\"i\\":\\"5\\"},\\"id\\":\\"178f61d0-47dc-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"6\\",\\"gridData\\":{\\"x\\":0,\\"y\\":3,\\"w\\":6,\\"h\\":3,\\"i\\":\\"6\\"},\\"id\\":\\"61b18a10-44d8-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"7\\",\\"gridData\\":{\\"x\\":6,\\"y\\":3,\\"w\\":6,\\"h\\":3,\\"i\\":\\"7\\"},\\"id\\":\\"6e3552d0-47f8-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"8\\",\\"gridData\\":{\\"x\\":0,\\"y\\":6,\\"w\\":6,\\"h\\":3,\\"i\\":\\"8\\"},\\"version\\":\\"6.2.2\\",\\"type\\":\\"visualization\\",\\"id\\":\\"04535f20-4801-11e8-8d68-419cfa041388\\"}]",
            "timeRestore": false,
            "title": "Host Dashboard",
            "version": 1,
          },
          "type": "dashboard",
          "updated_at": "2018-04-24T22:21:35.904Z",
        },
        "_type": "doc",
      },
      Object {
        "_id": "dashboard:3999b800-44cd-11e8-8d68-419cfa041388",
        "_index": ".kibana",
        "_score": 1,
        "_source": Object {
          "config": Object {
            "tr-checksum": "499dbfd06272af8f92a0ae164c92c8da4c2efaaf",
          },
          "dashboard": Object {
            "description": "",
            "hits": 0,
            "kibanaSavedObjectMeta": Object {
              "searchSourceJSON": "{\\"query\\":{\\"language\\":\\"lucene\\",\\"query\\":\\"\\"},\\"filter\\":[],\\"highlightAll\\":true,\\"version\\":true}",
            },
            "optionsJSON": "{\\"darkTheme\\":true,\\"hidePanelTitles\\":false,\\"useMargins\\":true}",
            "panelsJSON": "[{\\"panelIndex\\":\\"1\\",\\"gridData\\":{\\"x\\":6,\\"y\\":6,\\"w\\":6,\\"h\\":3,\\"i\\":\\"1\\"},\\"id\\":\\"61b18a10-44d8-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"2\\",\\"gridData\\":{\\"x\\":6,\\"y\\":9,\\"w\\":6,\\"h\\":3,\\"i\\":\\"2\\"},\\"id\\":\\"fca79400-44cf-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"3\\",\\"gridData\\":{\\"x\\":0,\\"y\\":12,\\"w\\":6,\\"h\\":3,\\"i\\":\\"3\\"},\\"id\\":\\"6e3552d0-47f8-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"4\\",\\"gridData\\":{\\"x\\":6,\\"y\\":12,\\"w\\":6,\\"h\\":3,\\"i\\":\\"4\\"},\\"id\\":\\"178f61d0-47dc-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"5\\",\\"gridData\\":{\\"x\\":0,\\"y\\":15,\\"w\\":6,\\"h\\":3,\\"i\\":\\"5\\"},\\"id\\":\\"04535f20-4801-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"6\\",\\"gridData\\":{\\"x\\":0,\\"y\\":0,\\"w\\":6,\\"h\\":3,\\"i\\":\\"6\\"},\\"id\\":\\"00ea4aa0-489d-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"7\\",\\"gridData\\":{\\"x\\":0,\\"y\\":3,\\"w\\":6,\\"h\\":3,\\"i\\":\\"7\\"},\\"id\\":\\"d9ac58a0-48a3-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"8\\",\\"gridData\\":{\\"x\\":6,\\"y\\":0,\\"w\\":6,\\"h\\":3,\\"i\\":\\"8\\"},\\"id\\":\\"59da5c10-48a0-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"9\\",\\"gridData\\":{\\"x\\":6,\\"y\\":3,\\"w\\":6,\\"h\\":3,\\"i\\":\\"9\\"},\\"id\\":\\"1a792fb0-48aa-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"10\\",\\"gridData\\":{\\"x\\":0,\\"y\\":6,\\"w\\":6,\\"h\\":3,\\"i\\":\\"10\\"},\\"id\\":\\"5740c040-48b2-11e8-8d68-419cfa041388\\",\\"type\\":\\"visualization\\",\\"version\\":\\"6.2.2\\"},{\\"panelIndex\\":\\"11\\",\\"gridData\\":{\\"x\\":0,\\"y\\":9,\\"w\\":6,\\"h\\":3,\\"i\\":\\"11\\"},\\"version\\":\\"6.2.2\\",\\"type\\":\\"visualization\\",\\"id\\":\\"311d6b30-48a7-11e8-8d68-419cfa041388\\"}]",
            "timeRestore": false,
            "title": "Robin's Dashboard",
            "version": 1,
          },
          "type": "dashboard",
          "updated_at": "2018-04-25T21:56:18.359Z",
        },
        "_type": "doc",
      },
    ],
  },
}
`;

exports[`getEsState filters search results and excludes config type hits 2`] = `
Object {
  "index": ".kibana",
  "size": 1000,
  "type": "doc",
}
`;

exports[`removeItem calls esClient with correct params 1`] = `
Object {
  "id": "index-pattern:9d8143d0-3ce2-11e8-80cc-950e99edcb04",
  "index": ".kibana",
  "type": "doc",
}
`;

exports[`updateItem calls esClient with correct params 1`] = `
Object {
  "body": Object {
    "config": Object {
      "tr-checksum": "5332d14e81762116caaf18a82ae1b1753ff91781",
    },
    "index-pattern": Object {
      "fields": "[{\\"name\\":\\"AA\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"RA\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"RD\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"TC\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"TTLs\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"_id\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":false},{\\"name\\":\\"_index\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":false},{\\"name\\":\\"_score\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":false,\\"aggregatable\\":false,\\"readFromDocValues\\":false},{\\"name\\":\\"_source\\",\\"type\\":\\"_source\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":false,\\"aggregatable\\":false,\\"readFromDocValues\\":false},{\\"name\\":\\"_type\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":false},{\\"name\\":\\"answers\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"auth\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"id.orig_h\\",\\"type\\":\\"ip\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"id.orig_p\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"id.resp_h\\",\\"type\\":\\"ip\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"id.resp_p\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"proto\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"qclass\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"qclass_name\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"qtype\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"qtype_name\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"query\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"rcode\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"rcode_name\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"rejected\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"saw_query\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"saw_reply\\",\\"type\\":\\"boolean\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"total_answers\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"total_replies\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"trans_id\\",\\"type\\":\\"number\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"ts\\",\\"type\\":\\"date\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true},{\\"name\\":\\"uid\\",\\"type\\":\\"string\\",\\"count\\":0,\\"scripted\\":false,\\"searchable\\":true,\\"aggregatable\\":true,\\"readFromDocValues\\":true}]",
      "timeFieldName": "ts",
      "title": "metadata_dnsrecordinfo*",
    },
    "type": "index-pattern",
    "updated_at": "2018-04-10T17:14:25.905Z",
  },
  "id": "index-pattern:9d8143d0-3ce2-11e8-80cc-950e99edcb04",
  "index": ".kibana",
  "type": "doc",
}
`;
